android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java

```diff
@@ -461,16 +460,12 @@ public static void setInputMethodService(InputMethodService service) {
 
   /**
    * Get the input connection based on the keyboard type.
-   * For InApp keyboard, save the connection to inappInputConnection
    * @param {KeyboardType} keyboard
    * @return InputConnection
    */
   protected static InputConnection getInputConnection(KeyboardType keyboard) {
     if (keyboard == KeyboardType.KEYBOARD_TYPE_INAPP) {
-      if (inappInputConnection == null) {
-        inappInputConnection = KMTextView.activeView.onCreateInputConnection(new EditorInfo());
-      }
-      return inappInputConnection;
+      return KMTextView.activeView.onCreateInputConnection(new EditorInfo());
```

Comment:
Is this going to potentially cause other problems with losing context in some apps? `onCreateInputConnection()` was not used anywhere in stable-16.0 that I can see, so this is  a new API usage?

Reply:
This was introduced in #8438 

> Convert's KMKeyboardJSHandler.insertText from an abstract method to strictly using SystemKeyboard's copy of `insertText`. ~I anticipated having to add more logic between the two, but the InApp Keyboard seems to still work.~
> (edit) - It turns out I had to add new code to get the InputConnection for InAppKeyboard.

This should only impact the Keyman app.

Reply:
OK. Do we need to run new user tests to verify that we won't lose e.g. deadkey state?

Reply:
Yeah I can add the tests from #8611... Will try locally first.